core-tests/src/test/java/org/mule/runtime/core/internal/util/store/PersistentObjectStorePartitionTestCase.java

```diff
@@ -100,11 +101,13 @@ public void clear() throws Exception {
 
 
     partition.store(KEY, VALUE);
-    assertTrue(partition.contains(KEY));
-    assertEquals(VALUE, partition.retrieve(KEY));
+    assertThat(partition.contains(KEY), is(true));
+    assertThat(VALUE, is(partition.retrieve(KEY)));
 
     partition.clear();
-    assertFalse(partition.contains(KEY));
+    assertThat(partition.contains(KEY), is(false));
```

Comment:
This is testing the implementation not the behavior. What you should try to reproduce is that without your fix, writing to a store after clear fails and that afterwards you can successfully write and read from a cleared object store